catalyst_voices/apps/voices/lib/pages/campaign/admin_tools/campaign_admin_tools_dialog.dart

```diff
@@ -0,0 +1,309 @@
+import 'package:catalyst_voices/common/ext/space_ext.dart';
+import 'package:catalyst_voices/pages/campaign/admin_tools/campaign_admin_tools_events.dart';
+import 'package:catalyst_voices/pages/campaign/admin_tools/campaign_admin_tools_views.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+/// A draggable [CampaignAdminToolsDialog],
+/// should be used as a child of a [Stack].
+///
+/// Initially shown at bottom-left corner with [initialPadding] offset.
+class DraggableCampaignAdminToolsDialog extends StatefulWidget {
+  /// The key for the [CampaignAdminToolsDialog] to make sure it's state
+  /// is kept when a user keeps dragging it.
+  ///
+  /// The state might include currently open tab from [TabBar],
+  /// scroll position, etc.
+  final GlobalKey dialogKey;
+
+  /// See [CampaignAdminToolsDialog.selectedSpace].
+  final Space selectedSpace;
+
+  /// See [CampaignAdminToolsDialog.onSpaceSelected].
+  final ValueChanged<Space> onSpaceSelected;
+
+  /// See [CampaignAdminToolsDialog.onClose].
+  final VoidCallback onClose;
+
+  /// The initial offset from bottom-left for the dialog.
+  ///
+  /// The widget should be used inside a stack so left/bottom properties
+  /// from [EdgeInsets] correspond to left/bottom properties
+  /// from the related [Positioned] widget.
+  final EdgeInsets initialPadding;
```

Comment:
In my opinion `Offset` would be better here since you're using only two properties, `left` and `bottom` and its confusing why other do nothing.

Reply:
Replaced with offset